cfscrape/__init__.py

```diff
@@ -152,20 +153,26 @@ def solve_cf_challenge(self, resp, **original_kwargs):
         body = resp.text
         parsed_url = urlparse(resp.url)
         domain = parsed_url.netloc
-        submit_url = "%s://%s/cdn-cgi/l/chk_jschl" % (parsed_url.scheme, domain)
+        if self.org_method is None:
+            self.org_method = resp.request.method
+        submit_url = "%s://%s/%s" % (parsed_url.scheme,
+                                     domain,
+                                     re.findall('\<form id=\"challenge-form\" action=\"\/(.*)\?', resp.text)[0])
 
         cloudflare_kwargs = copy.deepcopy(original_kwargs)
 
         headers = cloudflare_kwargs.setdefault("headers", {})
         headers["Referer"] = resp.url
 
         try:
-            params = cloudflare_kwargs["params"] = OrderedDict(
+            cloudflare_kwargs["data"] = OrderedDict(
                 re.findall(r'name="(s|jschl_vc|pass)"(?: [^<>]*)? value="(.+?)"', body)
             )
+            params = cloudflare_kwargs["params"] = {'__cf_chl_jschl_tk__': re.findall(
+                '\<form id=\"challenge-form\" action=\".*\?__cf_chl_jschl_tk__=(.*?)\"', resp.text)[0]}
```

Comment:
```suggestion
                '\<form id=\"challenge-form\" action=\".*(?:\?|\&)__cf_chl_jschl_tk__=(.*?)\"', resp.text)[0]}
```
On a request like https://kissanime.ru/Anime/Meiken-Lassie/Episode-024?id=16447, the action attrib is now `action="/Anime/Meiken-Lassie/Episode-024?id=164477&__cf_chl_jschl_tk__...` 
The first re pattern would not find  `__cf_chl_jschl_tk__` since there is an **&** before

Im not sure if thats intended or would break something else but works for me